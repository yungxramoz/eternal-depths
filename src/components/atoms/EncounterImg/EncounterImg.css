.encounter-img {
  filter: drop-shadow(-2px 2px 5px black);
}

.encounter-container {
  display: flex;
  flex-grow: 1;
  padding: 3em 0 1em 0;
}
