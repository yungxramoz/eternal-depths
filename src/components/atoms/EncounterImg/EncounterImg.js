import React from 'react'
import './EncounterImg.css'

const EncounterImg = ({ encounterAnimationState, encounter }) => {
  let encounterImgSrc = require('../../../assets/images/encounters/' +
    encounter.fileName)

  return (
    <div className="encounter-container">
      <img
        className={`encounter-img ${encounterAnimationState}`}
        src={encounterImgSrc}
        alt={encounter.name}
        style={encounter.style}
      />
    </div>
  )
}

export default EncounterImg
