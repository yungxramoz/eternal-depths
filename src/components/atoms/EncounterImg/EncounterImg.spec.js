import React from 'react'
import { render, screen } from '@testing-library/react'
import EncounterImg from './EncounterImg'

describe('EncounterImg', () => {
  const encounter = {
    name: 'Test Encounter',
    fileName: 'test_encounter.png',
    style: { width: '100px', height: '100px' },
  }

  const encounterAnimationState = 'test-animation-state'

  it('renders correctly', () => {
    render(
      <EncounterImg
        encounter={encounter}
        encounterAnimationState={encounterAnimationState}
      />,
    )
    expect(document.body).toMatchSnapshot('EncounterImg')
  })
})
