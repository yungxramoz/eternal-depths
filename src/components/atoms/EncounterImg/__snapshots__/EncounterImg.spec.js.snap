// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EncounterImg renders correctly: EncounterImg 1`] = `
<body>
  <div>
    <div
      class="encounter-container"
    >
      <img
        alt="Test Encounter"
        class="encounter-img test-animation-state"
        src="test-file-stub"
        style="width: 100px; height: 100px;"
      />
    </div>
  </div>
</body>
`;
